misc.go

```diff
@@ -65,22 +65,23 @@ func (e *RateLimitedError) Error() string {
 }
 
 func fileUploadReq(ctx context.Context, path, fieldname, filename string, values url.Values, r io.Reader) (*http.Request, error) {
-	body := &bytes.Buffer{}
-	wr := multipart.NewWriter(body)
-
-	ioWriter, err := wr.CreateFormFile(fieldname, filename)
-	if err != nil {
-		wr.Close()
-		return nil, err
-	}
-	_, err = io.Copy(ioWriter, r)
-	if err != nil {
-		wr.Close()
-		return nil, err
-	}
-	// Close the multipart writer or the footer won't be written
-	wr.Close()
-	req, err := http.NewRequest("POST", path, body)
+	pipeReader, pipeWriter := io.Pipe()
+	wr := multipart.NewWriter(pipeWriter)
+	go func() {
+		defer pipeWriter.Close()
+		ioWriter, err := wr.CreateFormFile(fieldname, filename)
+		if err != nil {
+			fmt.Println(err)
```

Comment:
these errors could be handled better. and certainly shouldn't just be dumped to stdout.

I'd look to move the pipe creation into `postWithMultipartResponse` so you can check for write errors after the request is processed.